### PR TITLE
Reduce logging during incremental mapping

### PR DESCRIPTION
**Pull request by sarlinpe**
IMHO regular users don't need to know about statistics during reconstruction. This PR changes some logs to verbose. To enable them:
- prefix the command with `GLOG_v=1`
- add the CLI argument `--v=1`
- or set `pycolmap.logging.verbose_level = 1` in Python

Example of default output (`python example.py`):
```
I20240213 23:45:11.468131 139676170342400 misc.cc:198]
==============================================================================
Finding good initial image pair
==============================================================================
I20240213 23:45:11.576729 139676170342400 misc.cc:198]
==============================================================================
Initializing with image pair #6 and #4
==============================================================================
I20240213 23:45:11.584307 139676170342400 misc.cc:198]
==============================================================================
Global bundle adjustment
==============================================================================
I20240213 23:45:12.467502 139676170342400 misc.cc:198]
==============================================================================
Registering image #5 (3)
==============================================================================
I20240213 23:45:12.467517 139676170342400 incremental_mapper.cc:496] => Image sees 3755 / 8696 points
I20240213 23:45:13.138308 139676170342400 misc.cc:198]
==============================================================================
Retriangulation
==============================================================================
I20240213 23:45:13.141402 139676170342400 misc.cc:198]
==============================================================================
Global bundle adjustment
==============================================================================
I20240213 23:45:13.533418 139676170342400 misc.cc:198]
==============================================================================
Registering image #7 (4)
==============================================================================
I20240213 23:45:13.533433 139676170342400 incremental_mapper.cc:496] => Image sees 4844 / 8946 points
I20240213 23:45:14.172859 139676170342400 misc.cc:198]
==============================================================================
Retriangulation
==============================================================================
I20240213 23:45:14.177424 139676170342400 misc.cc:198]
==============================================================================
Global bundle adjustment
==============================================================================
```

### CODE REVIEW

**Comment by sarlinpe** [comment]
We could consider removing the header formatting (downgrade `PrintHeading1` to `LOG(INFO)`) - @ahojnnes wdyt? This would properly show the call site in `incremental_mapper.cc` instead of `misc.cc`.

**Comment by ahojnnes** [comment]
I am fine with removing all the headings, if the more detailed information is hidden behind verbose logging.

**Comment by sarlinpe** [comment]
Done, new output
```
I20240214 16:20:29.593710 2679042 incremental_mapper.cc:402] Finding good initial image pair
I20240214 16:20:29.705181 2679042 incremental_mapper.cc:424] Initializing with image pair #6 and #4
I20240214 16:20:29.712884 2679042 incremental_mapper.cc:64] Global bundle adjustment
I20240214 16:20:30.609822 2679042 incremental_mapper.cc:492] Registering image #5 (3)
I20240214 16:20:30.609841 2679042 incremental_mapper.cc:495] => Image sees 3757 / 8695 points
I20240214 16:20:31.283254 2679042 incremental_mapper.cc:103] Retriangulation
I20240214 16:20:31.286366 2679042 incremental_mapper.cc:64] Global bundle adjustment
I20240214 16:20:31.678565 2679042 incremental_mapper.cc:492] Registering image #7 (4)
I20240214 16:20:31.678586 2679042 incremental_mapper.cc:495] => Image sees 4845 / 8945 points
I20240214 16:20:32.315026 2679042 incremental_mapper.cc:103] Retriangulation
I20240214 16:20:32.319654 2679042 incremental_mapper.cc:64] Global bundle adjustment
```

File: src/colmap/controllers/incremental_mapper.cc [hunk 1]
```diff
@@ -237,12 +236,12 @@ BundleAdjustmentOptions IncrementalMapperOptions::GlobalBundleAdjustment()
   options.solver_options.max_linear_solver_iterations = 100;
   options.solver_options.logging_type =
       ceres::LoggingType::PER_MINIMIZER_ITERATION;
-  options.solver_options.minimizer_progress_to_stdout = true;
+  options.solver_options.minimizer_progress_to_stdout = false;
```
Comment (S-o-T) [inline src/colmap/controllers/incremental_mapper.cc:239]: Please, consider not disabling minimizer progress printing if verbose output is requested.

Comment (ahojnnes) [reply src/colmap/controllers/incremental_mapper.cc:239]: Yes, this makes sense. I would agree.

Comment (ahojnnes) [reply src/colmap/controllers/incremental_mapper.cc:239]: There is VLOG_IS_ON(1) that can be used here.

Comment (sarlinpe) [reply src/colmap/controllers/incremental_mapper.cc:239]: This is already the case: if false, Ceres logs to VLOG(1) instead of stdout

Comment (sarlinpe) [reply src/colmap/controllers/incremental_mapper.cc:239]: We could log to stdout instead with VLOG_IS_ON, but then it doesn’t get written to additional sinks (eg file)

Comment (ahojnnes) [reply src/colmap/controllers/incremental_mapper.cc:239]: Got it. Thanks.

**Review by ahojnnes — APPROVED** [review]

**Pull request merged**